### PR TITLE
UCX: backports from 1.3/master

### PR DESCRIPTION
**Pull request by hoopoepg**
- backports of some fixes

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5139/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5144/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5148/ for details (Mellanox internal link).

**Comment by hoopoepg** [comment]
bot:mlx:retest

File: src/ucp/api/ucp_compat.h [hunk 1]
```diff
@@ -33,4 +35,56 @@ void ucp_request_release(void *request);
 void ucp_ep_destroy(ucp_ep_h ep);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ * @deprecated Replaced by @ref ucp_ep_close_nb.
+ */
+ucs_status_ptr_t ucp_disconnect_nb(ucp_ep_h ep);
```
Comment (yosefe) [inline src/ucp/api/ucp_compat.h]: why need to add functions to compat.h? should be just adding C wrapper..

Comment (hoopoepg) [reply src/ucp/api/ucp_compat.h]: really strange... when I cherry-pick this commit git displayed that there was indentation change only, no new lines.

I rebased commit again

**Comment by hoopoepg** [comment]
@yosefe not sure where from additional funcs are come. I just rebased commit "external C" and it gone

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5177/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5178/ for details (Mellanox internal link).

**Pull request merged**